application/src/main/java/org/togetherjava/tjbot/features/basic/CoolMessagesBoardManager.java

```diff
@@ -0,0 +1,129 @@
+        Optional<TextChannel> boardChannel = getBoardChannel(event.getJDA(), guildId);
+
+        if (boardChannel.isEmpty()) {
+            logger.warn("Tried to find the board channel in server {} but was unable to do so",
+                    guildId);
+            return;
+        }
+
+        // If the bot has already reacted to this message, then this means that
+        // the message has been quoted to the cool messages board, so skip it.
+        if (hasBotReacted(event.getJDA(), messageReaction)) {
+            return;
+        }
+
+        if (isCoolEmoji && originalReactionsCount + 1 >= config.minimumReactions()) {
+            event.retrieveMessage().queue(message -> {
+                message.addReaction(REACT_EMOJI).queue();
+
+                insertCoolMessage(boardChannel.orElseThrow(), message);
+            }, e -> logger.warn("Tried to retrieve cool message but got: {}", e.getMessage()));
+        }
+    }
+
+    /**
+     * Gets the board text channel where the quotes go to, wrapped in an optional.
+     *
+     * @param jda the JDA
+     * @param guildId the guild ID
+     * @return the board text channel
+     */
+    private Optional<TextChannel> getBoardChannel(JDA jda, long guildId) {
+        return jda.getGuildById(guildId)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> boardChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    /**
+     * Inserts a message to the specified text channel
+     */
+    private static void insertCoolMessage(TextChannel boardChannel, Message message) {
+        boardChannel.sendMessageEmbeds(Collections.singleton(createQuoteEmbed(message))).queue();
+    }
+
+    /**
+     * Wraps a text message into a properly formatted quote message used for the board text channel.
+     */
+    private static MessageEmbed createQuoteEmbed(Message message) {
+        final User author = message.getAuthor();
+        EmbedBuilder embedBuilder = new EmbedBuilder();
+
+        // If the message contains image(s), include the first one
+        var firstImageAttachment = message.getAttachments()
+            .stream()
+            .parallel()
+            .filter(Message.Attachment::isImage)
+            .findAny()
+            .orElse(null);
+
+        if (firstImageAttachment != null) {
+            embedBuilder.setThumbnail(firstImageAttachment.getUrl());
+        }
+
+        return embedBuilder.setDescription(message.getContentDisplay())
+            .appendDescription("%n%n[Jump to Message](%s)".formatted(message.getJumpUrl()))
+            .setColor(Color.orange)
+            .setAuthor(author.getName(), null, author.getAvatarUrl())
+            .setTimestamp(message.getTimeCreated())
+            .build();
+    }
+
+    /**
+     * Checks a {@link MessageReaction} to see if the bot has reacted to it.
+     */
+    private static boolean hasBotReacted(JDA jda, MessageReaction messageReaction) {
```

Comment:
Can you change this to check that the reaction was the emoji we expect? Mainly for future proofing if we ever make the bot react to messages for another feature.

Reply:
That's true! I have added an extra check where this method always returns `false` if the `messageReaction` is dealing with an emoji that is not the star emoji used here. 🌟